broker-daemon/state-machines/interchain.js

```diff
@@ -0,0 +1,310 @@
+ * @type {Number}
+ * @constant
+ */
+const DEFAULT_MIN_FINAL_DELTA = 86400
+
+/**
+ * The amount of time, in seconds, that we'd like to buffer any output timelock
+ * by to account for block ticks during a swap This is especially problematic on
+ * regtest where we mine blocks every 10 seconds and is a known issue on
+ * mainnet.
+ *
+ * @see {@link https://github.com/lightningnetwork/lnd/issues/535}
+ * @type {Number}
+ * @constant
+ */
+const BLOCK_BUFFER = 1200
+
+/**
+ * The minimum time lock (in seconds) on extended HTLCs in order for them to be
+ * accepted. This assumes a static route from this node, through the Relayer,
+ * to the receiving node.
+ *
+ * @todo Make this value dynamic to accept different routes
+ * and different forwarding policies / final cltv deltas
+ * @type {number}
+ * @constant
+ * @returns {number} Time delta in seconds
+ */
+const OUTBOUND_TIME_LOCK = DEFAULT_RELAYER_FWD_DELTA + DEFAULT_MIN_FINAL_DELTA + BLOCK_BUFFER
+
+/**
+ * The minimum time lock (in seconds) on inbound HTLCs for us to accept them and
+ * be able to forward them on.
+ *
+ * @todo Make this value dynamic to accept different routes
+ * and different forwarding policies / final cltv deltas
+ * @type {number}
+ * @constant
+ */
+const INBOUND_TIME_LOCK = OUTBOUND_TIME_LOCK + DEFAULT_MAKER_FWD_DELTA + BLOCK_BUFFER
+
+/**
+ * Prepare for a swap by setting up a hold invoice
+ * on the inbound chain.
+ *
+ * @public
+ *
+ * @param {string}  hash                  - Base64 string of the hash for the
+ *                                          swap
+ * @param {Payment} inboundPayment
+ * @param {Engine}  inboundPayment.engine - Engine of the expected inbound
+ *                                          payment
+ * @param {number}  inboundPayment.amount - Amount, in the smallest unit, of the
+ *                                          expected inbound payment.
+ * @param {Date}    timeout               - Absolute time after which the
+ *                                          payment should not be translated.
+ */
+async function prepareSwap (hash, { engine, amount }, timeout) {
+  await engine.prepareSwap(
+    hash,
+    amount,
+    INBOUND_TIME_LOCK,
+    timeout
+  )
+}
+
+/**
+ * Translate a swap payment to the other chain in an idempotent fashion,
+ * i.e. by only paying the downstream invoice if it is not available any
+ * other way. This is done by retrieving the preimage:
+ * - from an existing outbound payment
+ * - from an existing settled inbound invoice
+ * - by making a new outbound payment
+ *
+ * @private
+ *
+ * @param {string}  hash                    - Base64 string of the hash for the
+ *                                            swap
+ * @param {Payment} inboundPayment
+ * @param {Engine}  inboundPayment.engine   - Engine of the expected inbound
+ *                                            payment
+ * @param {Payment} outboundPayment
+ * @param {Engine}  outboundPayment.engine  - Engine of the outbound
```

Comment:
Since we have a `typedef` for `Payment` above, I think we can remove the redundant description here.